GazdApp/src/main/java/hu/elte/gazdapp/controller/MainController.java

```diff
@@ -9,124 +9,314 @@
+        	try {
+				updateThread.join(1000);
+			} catch (InterruptedException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+        }
     }
 
-    public void start() {
-        board.start();
-        gui.update();
+    public void startServer() {
+        try {
+            registry = LocateRegistry.createRegistry(12345);
+            Board remoteBoard = new Board();
+            registry.rebind("rmiServer", remoteBoard);
+            for(Player p : board.getPlayers()){
+               remoteBoard.addPlayer(p); 
+            }
+            board = remoteBoard;
+            board.start();
+            gui.update();
+            
+            
+            isServer = true;
+            communicationSocket = new MulticastSocket(54321);
+            InetAddress group = InetAddress.getByName("234.234.234.234");
+            communicationSocket.joinGroup(group);            
+            
+        } catch (RemoteException ex) {
+            Logger.getLogger(MainController.class.getName()).log(Level.SEVERE, null, ex);
+        } catch (IOException ex) {
+        	Logger.getLogger(MainController.class.getName()).log(Level.SEVERE, null, ex);
+		}
+    }
+    
+    public void startClient() {
+        try {
+            registry = LocateRegistry.getRegistry("localhost", 12345);
+            board = (BoardInterface) (registry.lookup("rmiServer"));
+            
+            gameInProgress = true;
+            
+            board.start();
+            gui.update();
+            
+            isServer = false;
+            communicationSocket = new MulticastSocket(54321);
+            InetAddress group = InetAddress.getByName("234.234.234.234");
+            communicationSocket.joinGroup(group);
+            
+            updateThread = new Thread(() -> {
+                try {
+	            	byte[] buf = new byte[256];
+	            	while (gameInProgress) {
+	                    DatagramPacket packet = new DatagramPacket(buf, buf.length);
+					    communicationSocket.receive(packet);	
+	                    String received = new String(
+	                      packet.getData(), 0, packet.getLength());
+	                    if ("update".equals(received)) {
+	                        gui.update();
+	                    } else {
+	                    	break;
+	                    }
+	                }
+                } catch (IOException ex) {
+                	Logger.getLogger(MainController.class.getName()).log(Level.SEVERE, null, ex);
+				}
+            });
+            updateThread.start();
+            
+        } catch (Exception ex) {
+            Logger.getLogger(MainController.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+        
+    public void updateClients() {
+    	if (!isServer) {
+    		return;
+    	}
+    	byte updateMsg[] = "update".getBytes();
+    	DatagramPacket updatePacket = new DatagramPacket(updateMsg, updateMsg.length);
+    	try {
+			communicationSocket.send(updatePacket);

```

Comment:
Does it work for you? Maybe I'm doing something wrong, but it dies here on roll with "java.lang.NullPointerException: null address || null buffer".